text/0005-config_design.md

```diff
@@ -0,0 +1,110 @@
+- Feature Name: Redesign of nushells config structure
+- Start Date: (fill me in with today's date, 2021-03-18)
+- RFC PR: [nushell/rfcs#0005](https://github.com/nushell/rfcs/pull/0005)
+
+# Summary
+[summary]: #summary
+
+This RFC shall provide the basis for a structured discussion, how the nushell config could be improved.
+
+# Motivation
+[motivation]: #motivation
+Important: This document is written as if, https://github.com/nushell/nushell/pull/3041 would already be accepted and merged. The PR is currently in review.
+
+
+Nushell evolved over time and features got added. The current config format may need to be adapted to better represent nushells capabilities to new users.
+
+The config structure is mostly well done, but some enhancements I can think of:
+1. Every `[...]` table has a name without config, but there is `[color_config]` which I think should be renamed to `theme` (better consistency, better naming).
+2. We could add a table for aliases. Example:
+```toml
+[aliases]
+vim = "vscode" # The new kids on the block won't even notice
+emacs = "echo Hit exit-meta-alt-control-shift at the same time to enter emacs" # Hehe does nothing :)
+```
+3. We could give $PATH ($nu.path) his own section. Example:
+```toml
+[path]
+defaults = ["/usr/bin", "/home/sweet/home"]
+vpn_client = "/opt/vpn_client/bin"
+cargo = "/home/leo/.cargo"
+```
+4. Some settings are not put inside a table `[...]`. Those could at least be grouped under `[general]`.
+5. With `.nu-env` and the "global" config under `$HOME/.config/nu/config.toml` being merged, we can think of `autoenv` as a local config (a config local to a directory). Therefore we could rename `.nu-env` to `.nu/config.toml`. (We may need to adapt the `autoenv` commands accordingly)
+    - Rename `autoenv trust` to `package trust` (see 9)
+    - Rename `autoenv untrust` to `package untrust` (see 9)
+6. We have `startup` and `on_exit`. We could rename `startup` to `on_enter` (Better naming, better consistency (`enter`/`exit` command)).
```

Comment:
We can also think about adding a `on_enter.nu` and `on_exit.nu` file right next to the config.toml. `on_enter` and `on_exit` config values don't scale past 1 line of code. Files (I think) would suit most users better.

Reply:
This reminds me, I've been looking at the `fish shell` source code recently. I noticed that most, if not all, of their built in commands have cpp code but they also all have *.fish commands too. i.e. there's `cd` an `cd.fish`.

Reply:
Sounds cool. Is it something we need to consider when redesigning the config (and how configs should work)?

Reply:
I think it's probably a broader conversation about `stdlib` and built-in commands in general. so, we can probably drop the discussion about it since it's not specifically related to config.

Reply:
Good. But I generally agree with your opinion. It would also allow us to prototype stuff faster :).